funding/commitment_type_negotiation.go

```diff
@@ -354,15 +368,17 @@ func implicitNegotiateCommitmentType(local,
 }
 
 // hasFeatures determines whether a set of features is supported by both the set
-// of local and remote features.
+// of local and remote features. The set of missing feature bits is also
+// returned.
 func hasFeatures(local, remote *lnwire.FeatureVector,
-	features ...lnwire.FeatureBit) bool {
+	features ...lnwire.FeatureBit) (bool, []lnwire.FeatureBit) {
```

Comment:
the only thing is that this is a mixture of the missing features... not sure how useful that is. So perhaps return 2 slices here? one for local and one for remote?

Reply:
I thought this too, but I think that this message is for debugging, so we can double-check what kind of feature we are sending with the one that it is inside the error message.

However, if you think that returns the `remote` and `local` features in the error message I can add it too, because it is a good idea anyway.

Waiting for your answer :)

Reply:
yes the thing we use the return values for happens to just be for logging, but I think we want the return values to be meaningful & useful in a code context too.

Reply:
Sounds good! So going to apply this change

Reply:
@ellemouton just keeping the history clean to review I added a `fixup` commit with the implementation of what you are suggesting https://github.com/lightningnetwork/lnd/pull/8740/commits/3c4caadf18cdd2e54a2155b1095c3da76507a022

let me know if you have a better idea on how to implement it